tests/e2e/restart_test.go

```diff
@@ -20,97 +18,141 @@
 
-	out, err = dockercli.KillDocker(s.hostIP)
+	// 3. Restart docker
+	out, err = dockercli.KillDocker(s.config.DockerHosts[1])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// 4. Verify detached status. Volume should be detached (within the timeout)
+	status = verification.VerifyDetachedStatus(s.volumeName, s.config.DockerHosts[1], s.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is still attached", s.volumeName))
+
+	out, err = dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerName)
 	c.Assert(err, IsNil, Commentf(out))
 
-	status = verification.VerifyDetachedStatus(s.volumeName, s.hostIP, s.esxIP)
+	// 5. Verify a container can be started to use the same volume on another host
+	out, err = dockercli.AttachVolume(s.config.DockerHosts[0], s.volumeName, s.containerName)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// 6. Restart the docker daemon on the other host
+	out, err = dockercli.KillDocker(s.config.DockerHosts[0])
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Clean up the container on host0
+	out, err = dockercli.RemoveContainer(s.config.DockerHosts[0], s.containerName)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// 7. Verify detached status for the volume
+	status = verification.VerifyDetachedStatus(s.volumeName, s.config.DockerHosts[1], s.config.EsxHost)
 	c.Assert(status, Equals, true, Commentf("Volume %s is still attached", s.volumeName))
 
-	log.Printf("END: restart_test.TestVolumeStaleMount")
+	// 8. Verify a container can be started to use the same volume on the original host
+	out, err = dockercli.ExecContainer(s.config.DockerHosts[1], s.volumeName, s.containerName)
+	c.Assert(err, IsNil, Commentf(out))
+
+	misc.LogTestEnd(c.TestName())
 }
 
-// Test vDVS plugin restart
+// TestPluginKill - Test vDVS plugin restart
 // 1. Attach a volume with restart=always flag
 // 2. Verify volume attached status
 // 3. Kill vDVS plugin
 // 4. Stop the container
 // 5. Start the same container
 // 6. Stop the container
 // 7. Verify volume detached status
-func (s *PluginSuite) TestPluginKill(c *C) {
-	log.Printf("START: restart_test.TestPluginKill")
+func (s *RestartTestData) TestPluginKill(c *C) {
+	misc.LogTestStart(c.TestName())
 
-	out, err := dockercli.AttachVolumeWithRestart(s.hostIP, s.volumeName, s.containerName)
+	// 1. Attach a volume with restart=always flag
+	out, err := dockercli.AttachVolumeWithRestart(s.config.DockerHosts[1], s.volumeName, s.containerName)
 	c.Assert(err, IsNil, Commentf(out))
 
-	status := verification.VerifyAttachedStatus(s.volumeName, s.hostIP, s.esxIP)
+	// 2. Verify volume attached status
+	status := verification.VerifyAttachedStatus(s.volumeName, s.config.DockerHosts[1], s.config.EsxHost)
 	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", s.volumeName))
 
-	out, err = dockercli.KillVDVSPlugin(s.hostIP)
+	// 3. Kill vDVS plugin
+	out, err = dockercli.KillVDVSPlugin(s.config.DockerHosts[1])
```

Comment:
Correct.  Hence the kill signal to plugin for have forceful and ungraceful kill.